File: 2.1/micro/python/relax.py
def relax(old: float, du: float, w: float) -> float:
    alt = du + w
    return alt if alt < old else old


def step(old: float, du: float, w: float) -> float:
    return relax(old, du, w)


if __name__ == "__main__":
    print(f"{step(10.0, 3.0, 4.0):.2f}")

File: 2.1/micro/haskell/Relax.hs
module Main where

relax :: Double -> Double -> Double -> Double
relax old du w =
  let alt = du + w
   in if alt < old then alt else old

step :: Double -> Double -> Double -> Double
step old du w = relax old du w

main :: IO ()
main = print (step 10.0 3.0 4.0)
